### DISCUSSION TITLE
Best Practices for open source builds targeting Homebrew installed libraries on M1 and Intel

### DISCUSSION THREAD

**Discussion by jspiro** [Homebrew/discussions#2477] [Everyday usage]
I'm a maintainer of GitX and we have hundreds of references in our third party dependencies to /usr/local for libraries, including in our precompiled binary, which doesn't start on an M1 out of the box. Even if we install dependencies like openssl, they're not found at `/opt/homebrew`, and the binary crashes on load:

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: /usr/local/opt/openssl@1.1/lib/libcrypto.1.1.dylib
```

What are the best practices for build systems and binaries to support Intel and M1 when dependent on Homebrew? I'm assuming the change was made with a solution in mind, but I cannot find discussion of this anywhere. I don't even know where to begin to fix this.

Trying to avoid symlinks and shims in the old location (IF that would even work!), because my understanding is the intention here is Intel goes to `/usr`, and ARM to `/opt` (PS: where can I learn more about this standard?), so this would be against that.

Cheers!

**Comment by SMillerDev** [comment]
I think the best solution would be to use a standard like pkg-config and CMake to let the system determine where dependencies live. Any reliance on specific homebrew paths only works if users have the exact same setup you expect.

**Reply by jspiro** [reply]
While that makes sense, is that gonna work for Xcode projects? I'm not an Xpert...

￼I agree paths should not be hard-coded, but they have been for many years in many projects I've used, so use is a de facto standard location now, which homebrew changed. That's fine, I just don't see a clear architectural solution for existing use cases.

I hope I'm explaining this clearly. Maybe I'm missing something obvious.

**Reply by SMillerDev** [reply]
`/usr/local` is not just a Homebrew path, it's a Unix standard for files. That's why Homebrew initially chose it and why it's often hardcoded into projects.

> While that makes sense, is that gonna work for Xcode projects? I'm not an Xpert...

It should but that depends on how you include things with XCode I think.
